Mathlib/LinearAlgebra/QuadraticForm/Basis.lean

```diff
@@ -18,7 +18,43 @@ open LinearMap (BilinMap)
 
 namespace QuadraticMap
 
-variable {ι R M N} [LinearOrder ι]
+section
+
+variable {ι R M N}
+
+variable [CommRing R] [AddCommGroup M] [AddCommGroup N] [Module R M] [Module R N] [DecidableEq ι]
+
+open Finsupp in
+theorem map_finsuppSum (Q : QuadraticMap R M N) (f : ι →₀ R) (g : ι → R → M) :
+    Q (f.sum g) =
+      f.sum (fun i r => Q (g i r)) +
+      ∑ p ∈ f.support.sym2 with ¬ p.IsDiag,
+        p.lift ⟨fun i j => polar Q (g i (f i)) (g j (f j)), fun _ _ => polar_comm _ _ _⟩ :=
+  QuadraticMap.map_sum _ _ _
+
+open Finsupp in
+theorem map_linearCombination (Q : QuadraticMap R M N) {g : ι → M} (l : ι →₀ R) :
+    Q (linearCombination R g l) =
+      (l.sum fun i r => (r * r) • Q (g i)) +
+      ∑ p ∈ l.support.sym2 with ¬ p.IsDiag,
+        p.lift
+          ⟨fun i j => (l i * l j) • polar Q (g i) (g j), fun i j => by
+            simp only [polar_comm, mul_comm]⟩ := by
+  simp_rw [linearCombination_apply, map_finsuppSum, polar_smul_left, polar_smul_right, map_smul,
+    mul_smul]
+
+theorem basis_expansion (Q : QuadraticMap R M N) (bm : Basis ι R M) (x : M) :
```

Comment:
Do we have this theorem for `LinearMap`? I would expect we do, and if so we should try to make the naming match.

Reply:
The bilinear version is `LinearMap.sum_repr_mul_repr_mul`

Reply:
So something like `QuadraticMap.sum_repl_sq_add_sum_repl_mul_polar` with the `Q x` on the RHS?